core/os/os.h

```diff
@@ -328,6 +328,16 @@ class OS {
 
 	virtual PreferredTextureFormat get_preferred_texture_format() const;
 
+	enum CompositingWindowManager {
+		COMPOSITING_WINDOW_MANAGER_NONE,
+		COMPOSITING_WINDOW_MANAGER_UNKNOWN,
+		COMPOSITING_WINDOW_MANAGER_DESKTOP_WINDOW_MANAGER,
+		COMPOSITING_WINDOW_MANAGER_QUARTZ_COMPOSITOR,
+		COMPOSITING_WINDOW_MANAGER_X11,
+		COMPOSITING_WINDOW_MANAGER_WAYLAND
```

Comment:
This means that core knows about all of these window managers, which makes the window manager code no longer self-contained to the platform folders.

This would make more sense as a String-based enum, since that allows for an API that can be added onto arbitrarily by any platform's code.